src/Chapter2.hs

```diff
@@ -783,11 +791,12 @@ mastered the skill of eta-reducing.
 divideTenBy :: Int -> Int
 divideTenBy x = div 10 x
 
--- TODO: type ;)
+listElementsLessThan :: Ord a => a -> [a] -> [a]
 listElementsLessThan x l = filter (< x) l
 
 -- Can you eta-reduce this one???
-pairMul xs ys = zipWith (*) xs ys
+pairMul :: Num a => [a] -> [a] -> [a]
+pairMul xs ys = zipWith (*) xs ys 
```

Comment:
Do you need help with eta-reduction?
I can elaborate more on the topic if needed 👍🏼

Reply:
I would very much like to hear how this one was supposed to be eta-reduced. Thank you guys so much for making this project. I've been having a blast with it.

Reply:
Thanks for your kind words 🤗 

Let's see what eta-reduction really means.
When eta-reducing, you need to remove arguments from both sides of the `=` sign. So the correct solution for the first example is:

```haskell
divideTenBy = div 10
```

See how it compares with the original code:

```diff
-divideTenBy x = div 10 x
+divideTenBy = div 10
```

The motivation behind eta-reduction if that if you have two functions `foo` and `bar` of the same type, and they return the same results for any arguments (so `foo x` always equal to `bar x` for any `x`). In such situations, we say that functions `foo` and `bar` are equal, and if we want to define `foo` in terms of `bar`, we can write `foo = bar` instead of a longer `foo x = bar x` (which means that `foo` returns the same result as `bar` for any `x`).

This function is a bit trickier, as actually, you can remove two vars at once. Let's start with one
```diff
-pairMul xs ys = zipWith (*) xs ys
+pairMul xs = zipWith (*) xs
```
as `pairMul xs` function returns the same that `zipWith (*) xs` at all the arguments.
But actually looking at this option, we see that we can do this again as `pairMul` returns the same results on any input as `zipWith (*)`:
```diff
-pairMul xs = zipWith (*) xs
+pairMul = zipWith (*)
```

Hope it makes a bit more sense now 🙂

Reply:
Nice! Thank you, that makes a lot of sense.